proto/api/v1/cloud_source_service.proto

```diff
@@ -0,0 +1,155 @@
+syntax = "proto3";
+
+option go_package = "v1";
+option java_package = "io.stackrox.proto.api.v1";
+
+import weak "google/api/annotations.proto";
+import "api/v1/empty.proto";
+import "api/v1/pagination.proto";
+import "gogoproto/gogo.proto";
+
+package v1;
+
+// CloudSource is an integration which provides a source for discovered
+// clusters.
+message CloudSource {
+    string id                            = 1;
+    string name                          = 2;
+    string type                          = 3;
```

Comment:
Let's make this a strongly typed enum, otherwise we fall in the same short-comings we experience with e.g. image integrations where consts are defined "manually" and have the potential of being duplicated.

Reply:
@pedrottimark Assuming this would require another mapping on the front end side? I guess that's alright?

Reply:
Good question. I needed to ask about `type` property. If I understand correctly, `type` value in other integrations is consistent with key of config property, if it exists. Compare:
https://github.com/stackrox/stackrox/blob/master/proto/storage/image_integration.proto#L12-L33
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Integrations/utils/integrationsList.ts#L80-L159

To extend the current frontend pattern:
1. So-called `source` that is **loosely** coupled to backend might be either `'cloudSources'` or `'cloudsources'` depending on hyphen or not.
2. The `type` values that are **tightly** coupled to backend would be `'paladinCloud'` or `'ocm'` because there is a config property. For your info, frontend uses `type` in several ways:
    * descriptors for cards on /main/integrations page
    * descriptors for columns of table if you click a card
    * as in comment, presentation strings for integration forms (however, if discovered clusters table has an integration column, integration **name** seems more specific than integration **type**)

Is that the way you think of it, Stephan and Daniel?

Reply:
Yes correct, `type` is strongly coupled to the types of `Config`. It's essentially a string/enum representation of it.